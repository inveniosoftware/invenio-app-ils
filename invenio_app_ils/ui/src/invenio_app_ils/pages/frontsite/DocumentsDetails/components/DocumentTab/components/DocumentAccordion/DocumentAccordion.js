import React, { Component } from 'react';
import { Icon, Accordion } from 'semantic-ui-react';
import { isEmpty } from 'lodash';

export default class DocumentAccordion extends Component {
  state = { activeIndex: 0 };

  toggleAccordion = (e, titleProps) => {
    const { index } = titleProps;
    const { activeIndex } = this.state;
    const newIndex = activeIndex === index ? -1 : index;

    this.setState({ activeIndex: newIndex });
  };

  render() {
    const { activeIndex } = this.state;
    const { documentMetadata } = this.props;
    return (
      <Accordion fluid styled data-test={documentMetadata.pid}>
        <Accordion.Title
          active={activeIndex === 0}
          index={0}
          onClick={this.toggleAccordion}
        >
          <Icon name="dropdown" />
          Abstract
        </Accordion.Title>
        <Accordion.Content active={activeIndex === 0}>
          {!isEmpty(documentMetadata.abstracts)
            ? documentMetadata.abstracts.map((abstract, index) => (
                <p key={index}>{abstract}</p>
              ))
            : null}
        </Accordion.Content>

        <Accordion.Title
          active={activeIndex === 1}
          index={1}
          onClick={this.toggleAccordion}
        >
          <Icon name="dropdown" />
          Information
        </Accordion.Title>
        <Accordion.Content active={activeIndex === 1}>
          {!isEmpty(documentMetadata.tags)
            ? documentMetadata.tags.map(tag => <p key={tag.pid}>{tag.name}</p>)
            : null}
        </Accordion.Content>

        <Accordion.Title
          active={activeIndex === 2}
          index={2}
          onClick={this.toggleAccordion}
        >
          <Icon name="dropdown" />
          Chapters
        </Accordion.Title>
        <Accordion.Content active={activeIndex === 2}>
          {!isEmpty(documentMetadata.chapters)
            ? documentMetadata.chapters.map((chapter, index) => (
                <p key={index}>{chapter}</p>
              ))
            : null}
        </Accordion.Content>
      </Accordion>
    );
  }
}
