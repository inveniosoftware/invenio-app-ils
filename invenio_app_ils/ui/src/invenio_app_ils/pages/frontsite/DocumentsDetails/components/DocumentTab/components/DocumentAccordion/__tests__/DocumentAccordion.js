import React from 'react';
import { mount } from 'enzyme';
import DocumentAccordion from '../DocumentAccordion';

describe('DocumentAccordion tests', () => {
  let component;
  afterEach(() => {
    if (component) {
      component.unmount();
    }
  });

  const documentMetadata = {
    _access: {},
    eitems: {
      total: 1,
      hits: [
        {
          description: 'Non ipsum',
          pid: '145',
          open_access: true,
        },
      ],
    },
    abstracts: [],
    authors: ['Est sit magnam quiquia.'],
    booklinks: [],
    chapters: [],
    circulation: {},
    pid: '71',
    document_type: ['BOOK'],
    files: [],
    keyword_pids: ['24'],
    keywords: [],
    languages: ['el', 'ro', 'fr', 'it', 'es'],
    publishers: ['Quiquia amet consectetur velit.'],
    related_records: [],
    series: {},
  };

  it('should render the document correctly', () => {
    component = mount(
      <DocumentAccordion documentMetadata={documentMetadata} />
    );
    expect(component).toMatchSnapshot();

    const rows = component
      .find('DocumentAccordion')
      .find('Accordion')
      .filterWhere(
        element => element.prop('data-test') === documentMetadata.pid
      );
    expect(rows).toHaveLength(1);
  });
});
