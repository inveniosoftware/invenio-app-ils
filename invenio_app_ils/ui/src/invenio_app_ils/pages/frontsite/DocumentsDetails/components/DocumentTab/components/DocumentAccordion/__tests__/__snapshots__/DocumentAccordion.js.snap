// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentAccordion tests should render the document correctly 1`] = `
<DocumentAccordion
  documentMetadata={
    Object {
      "_access": Object {},
      "_computed": Object {
        "eitems": Array [
          Object {
            "description": "Non ipsum",
            "open_access": true,
            "pid": "145",
          },
        ],
      },
      "abstracts": Array [],
      "authors": Array [
        "Est sit magnam quiquia.",
      ],
      "booklinks": Array [],
      "chapters": Array [],
      "circulation": Object {},
      "document_types": Array [
        "BOOK",
      ],
      "files": Array [],
      "keyword_pids": Array [
        "24",
      ],
      "keywords": Array [],
      "languages": Array [
        "el",
        "ro",
        "fr",
        "it",
        "es",
      ],
      "pid": "71",
      "publishers": Array [
        "Quiquia amet consectetur velit.",
      ],
      "related_records": Array [],
      "series": Object {},
    }
  }
>
  <Accordion
    data-test="71"
    fluid={true}
    styled={true}
  >
    <AccordionAccordion
      className="ui fluid styled"
      data-test="71"
      exclusive={true}
    >
      <div
        className="accordion ui fluid styled"
        data-test="71"
      >
        <AccordionTitle
          active={true}
          index={0}
          onClick={[Function]}
        >
          <div
            className="active title"
            onClick={[Function]}
          >
            <Icon
              as="i"
              name="dropdown"
            >
              <i
                aria-hidden="true"
                className="dropdown icon"
              />
            </Icon>
            Abstract
          </div>
        </AccordionTitle>
        <AccordionContent
          active={true}
        >
          <div
            className="content active"
          />
        </AccordionContent>
        <AccordionTitle
          active={false}
          index={1}
          onClick={[Function]}
        >
          <div
            className="title"
            onClick={[Function]}
          >
            <Icon
              as="i"
              name="dropdown"
            >
              <i
                aria-hidden="true"
                className="dropdown icon"
              />
            </Icon>
            Information
          </div>
        </AccordionTitle>
        <AccordionContent
          active={false}
        >
          <div
            className="content"
          />
        </AccordionContent>
        <AccordionTitle
          active={false}
          index={2}
          onClick={[Function]}
        >
          <div
            className="title"
            onClick={[Function]}
          >
            <Icon
              as="i"
              name="dropdown"
            >
              <i
                aria-hidden="true"
                className="dropdown icon"
              />
            </Icon>
            Chapters
          </div>
        </AccordionTitle>
        <AccordionContent
          active={false}
        >
          <div
            className="content"
          />
        </AccordionContent>
      </div>
    </AccordionAccordion>
  </Accordion>
</DocumentAccordion>
`;
